NHSDigital/validated-relationships-service-api
# NPA-3543: add initial /Consent endpoint and associated schema refactoring

## Ticket Link
https://nhsd-jira.digital.nhs.uk/browse/NPA-3543

## Description/Change Summary
These changes to the OAS cover the new `/Consent` API endpoint required for requesting proxy relationships from VRS. For each proxy relationship, there should be three resources:

- Consent (filtering possible based on `status`)
- RelatedPerson (optional _include in the bundle)
- Patient (optional _include in the bundle)

Draft requirements can be found in confluence here (along with details of how to map NAM payloads to FHIR resources): https://nhsd-confluence.digital.nhs.uk/display/NPA/Consent+endpoint+requirements

A summary of expected behaviour of the endpoint:

- Take IM1 connection params in HTTP header in order for VRS to use these against the new NAM endpoint outlined [here](https://nhsd-confluence.digital.nhs.uk/display/NO/Proxy%3A+Extending+the+NAM+to+return+Linked+Profiles+-+NHS+APP+-+Solution+Design). Design decision: I've decided to pass these params in a single delimited string that match the regex in order to be explicit about the expectation that these two values should be passed atomically
- If the IM1 connection params header is not present then VRS should only check it's local cache
- In case of a cache miss then we should hit the NAM for linked profiles and map these to FHIR resources before caching them and returning the resources requested (in accordance with `status` and `_include` query string parameter filters)
- The `performer:identifier` parameter should be used for querying the local cache
- The `status` parameter should be used for filtering the relationships (based on the `consent.status` property
- The `_include` parameter should be used for including referenced resources

# How to test?
No behaviour has been implemented as part of this change. Changes best reviewed in a swagger editor in addition to the raw changes

Of special note, ensure the previous two commits have been correctly incorporated into the spec due to these being merged manually.

- https://github.com/NHSDigital/validated-relationships-service-api/pull/101/files
- https://github.com/NHSDigital/validated-relationships-service-api/pull/104/files



## Review Checklist
:information_source: This section is to be filled in by the **reviewer**.
